pkg/prometheus/operator.go

```diff
@@ -1428,6 +1601,56 @@ func (c *Operator) selectServiceMonitors(p *monitoringv1.Prometheus) (map[string
 		})
 	}
 
+	// If denied by Prometheus spec, filter out all service monitors that access
+	// the file system.
+	if p.Spec.ArbitraryFSAccessThroughSMs.Deny {
+		for namespaceAndName, sm := range res {
+			for _, endpoint := range sm.Spec.Endpoints {
+				if endpoint.BearerTokenFile != "" {
```

Comment:
Maybe it would become more readable if we would move everything within this inner for loop into a separate func where we could use early returns instead of `continue`. No hard feelings though.

Reply:
That is a good idea. I moved it into a separate function also deduplicating the logging calls.